backend/main/src/User/Domain/User.ts

```diff
@@ -31,20 +25,28 @@ export class User extends AggregateRoot<string, IUserProps> {
 		return this.props.name;
 	}
 
-	get isAdmin(): boolean {
-		return this.props.isAdmin;
-	}
-
-	public changeUserName(name: string): User {
-		return new User(this.id, {
-			name,
-			isAdmin: false,
-		});
+	/**
+	 * change user's name.
+	 * @param name
+	 * @return User
+	 */
+	public changeUserName(name: string): Result<User, UserDomainError> {
+		return User.create(this.id, { name });
 	}
 }
 
+/**
+ * This class is the user's ID.
+ */
 export class UserId extends Identifier<string> {
 	static create(id: string): Result<UserId, UserIdDomainError> {
+		const regId =
+			/^[a-z 0-9]{8}-[a-z 0-9]{4}-[a-z 0-9]{4}-[a-z 0-9]{4}-[a-z 0-9]{12}$/;
+
+		if (!regId.test(id)) {
+			return Err(new UserIdDomainError("Incorrect User ID pattern"));
```

Comment:
Thank you for adding a Domain Logic to the UserId class! I found it causes some errors in the tests. Please check and fix them!